### PR TITLE
feat(eav, graphql): cache customAttributeMetadata

### PR DESCRIPTION
**Pull request by danslo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/magento2-custom-attribute-metadata-cache/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
`customAttributeMetadata` is uncached.

Fixes: N/A

## What is the new behavior?
This adds the functionality of this package, it caches `customAttributeMetadata`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

The graphql reader is just so that I don't have to copy stuff from core and then only add a cache identity. 

I chose this over modifying `schema.graphqls` so that it does not mess with other potential modifications to this query (maybe someone else adding a parameter or changing the return type of the query, etc)

### CODE REVIEW

File: Model/GraphQlReader.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Graycore\CustomAttributeMetadataCache\Model;
+
+use Magento\Framework\Config\ReaderInterface;
+
+class GraphQlReader implements ReaderInterface
+{
+    public function read($scope = null)
```
Comment (damienwebdev) [inline Model/GraphQlReader.php:9]: Why this change instead of a modification to `schema.graphqls`?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: Model/GraphQlReader.php (hunk 1 above)
Comment (danslo) [reply Model/GraphQlReader.php:9]: danslo — Today at 1:37 PM
the graphql reader is just so that I don't have to copy stuff from core and then only add a cache identity. that's mess with other potential modifications to this query (maybe someone else adding a parameter or changing the return type of the query, etc)
You're free to change it to use schema.graphqls, the "hard" stuff is in the CacheIdentity class 😛
and I pushed a change for the unit test
Candle — Today at 1:41 PM
Ah, ok. I like your version.

**Comment by damienwebdev** [comment]
Replaced by #2
